Flake8 Style Fixes

alanhdu opened the pull request:
This is a bunch of style fixes, so that we can turn `flake8` unconditionally on instead of using `flake8 --exit-zero`.

This PR fixes E200-299, which are all the whitespace pep8 issues. Let me know if this is worthwhile (I know some projects don't like big-bang style clean-ups) -- if it is, I can clean up all the other flake8 issues too.

kalefranz commented on conda/cli/find_commands.py:
```diff
@@ -25,7 +25,7 @@ def find_executable(executable, include_others=True):
     for dir_path in dir_paths:
         if sys.platform == 'win32':
-            for ext in  '.exe', '.bat', '':
+            for ext in {'.exe', '.bat', ''}:
```
Any reason not to just keep this a tuple?

    for ext in ('.exe', '.bat', ''):

kalefranz commented on conda/cli/main_package.py:
```diff
@@ -72,7 +72,7 @@ def remove(prefix, files):
         try:
             os.rmdir(path)
-        except OSError: # directory might not be empty
+        except OSError:     # directory might not be empty
```
Should only be 2 spaces, right? (instead of 5)

kalefranz commented on conda/cli/pscheck.py:
```diff
@@ -35,7 +35,7 @@ def check_processes(dir=root_dir, verbose=True):
     curpid = os.getpid()
     for n in psutil.get_pid_list():
-        if n == curpid: # The Python that conda is running is OK
+        if n == curpid:     # The Python that conda is running is OK
```
2 spaces instead of 4?

kalefranz commented on conda/compat.py:
```diff
@@ -55,13 +51,6 @@ def lchmod(path, mode):
-    def ceil(x):
-        from math import ceil
-        return int(ceil(x))
-    from pipes import quote
```
Are these removals no longer used in the project?  Even if, removal still concerns me because other repos `import conda`.  Use `# NOQA` if you have to.

kalefranz commented on conda/compat.py:
```diff
@@ -205,3 +194,6 @@ def __new__(cls, name, this_bases, d):
     return metaclass("NewBase", None, {})
+
+# Silence flake8
+(StringIO, urlparse, urllib_quote, zip, zip_longest)
```
Same here. Can we just use `# NOQA` to silence flake8?

alanhdu commented:
Made the changes. 

Yeah, `log2`, `ceil`, `configparser`, didn't show up in a git grep. Are you worried about them being used in other conda projects like `conda-env` or by 3rd-party projects?

For the unused import warnings in `conda/compat.py`, we can also think about turning flake8 off just for that file. That we, we can avoid a bunch of `# noqa` for unused imports and undefined names.

Also, should I go ahead and clean up all the other flake8 warnings here, or should I do that in a separate PR?

mcg1969 commented:
This is great, @alanhdu, thanks. It's not my responsibility to approve or merge but I do think we all benefit from cleaner code.

kalefranz commented:
> Are you worried about them being used in other conda projects like conda-env or by 3rd-party projects?

Yes, exactly.  `conda-env`, `conda-build`, and other projects where `conda` is being used as a python library.

> For the unused import warnings in conda/compat.py, we can also think about turning flake8 off just for that file. That way, we can avoid a bunch of # noqa for unused imports and undefined names.

I'm fine with that.  The `setup.cfg` file is where that goes.

> Also, should I go ahead and clean up all the other flake8 warnings here, or should I do that in a separate PR?

I'm fine with either :)

alanhdu commented on tests/test_misc.py:
```diff
@@ -28,5 +30,43 @@ def test_url_pat_3(self):
+        "testfile1": None,
+    }
+
+    make_mock_directory(tmpdir, mock_directory)
```
I couldn't find existing testing mocks for fake file systems, so I rolled my own with the built-in pytest tempdir fixture. Is this alright, or is there something else I should use?

kalefranz replied:
Fine as far as I'm concerned.  Actually great.  We're only at ~50% coverage right now.  The more tests the better.

alanhdu commented on setup.cfg:
```diff
@@ -14,13 +14,13 @@ markers =
 max-line-length = 99
 ignore = E121,E123,E126,E133,E226,E241,E242,E302,E704
-exclude = build/*,.tox/*,tests/*,ve/*,*/_vendor/*,auxlib/*,conda/progressbar/*,conda/compat.py
+exclude = build/*,.tox/*,tests/*,ve/*,*/_vendor/*,auxlib/*,conda/progressbar/*,conda/compat.py,utils/win_batlink.py
```
What exactly is `utils/win_batlink.py` for? The very first line is `# UNUSED MODULE`.

alanhdu commented on setup.cfg:
```diff
@@ -13,13 +13,13 @@ markers =
 [flake8]
 max-line-length = 99
-ignore = E121,E123,E126,E133,E226,E241,E242,E704
+ignore = E121,E123,E126,E133,E226,E241,E242,E302,E704
```
I decided to disable E302 (expected two blank lines instead of 1) instead of going through and fixing them all. There were a _ton_ of warnings (>100), and I don't really think it helps readability.

If you'd prefer to keep it on, though, I can go back and insert all the new lines.

alanhdu commented on tests/test_utils.py:
```diff
@@ -113,3 +113,31 @@ def test_stops_checking_as_soon_as_the_first_file_fails(self):
+
+        # ("z:\\miniconda\\",
+        #  "/z/miniconda/",
+        #  "/cygdrive/z/miniconda/"),
```
I added these test cases to try and figure out what the giant regexes from `win_path_to_unix` and `unix_path_to_win` were doing. I'm a little surprised that these cases fail. Is this intended behavior?

kalefranz replied:
@msarahan or @mingwandroid Can you comment here?

alanhdu commented:
> I'm fine with either :)

Let's just get it over with, then. The faster we get it done, the fewer merge conflicts we'll have to deal with. (Working on this definitely beats studying for my exams too :smile:)

If Travis passes this should be good to go on my end (modulo code review, of course). Not sure what's going on with the Appveyor build though...

alanhdu commented:
> Yes, exactly. conda-env, conda-build, and other projects where conda is being used as a python library.

Two questions about this: how does integration testing between all the different repos work? We could maybe clone the `conda-env` and `conda-build` repos and running their test suites as part the travis build here (maybe?) if there isn't any.

Also, is there a formal deprecation policy? There were a couple of unused functions, keyword arguments, and files, and I was wondering what the policy is for removing them?

kalefranz commented:
>  how does integration testing between all the different repos work?

It doesn't. They're coupled in a really horrible way when it comes to this.  Definitely have plans to fix this, but it won't happen immediately. Will take a bit of time.

> is there a formal deprecation policy

Nope. If it's unused, (this repo's) tests pass, and nobody raises an objection with the code review, get rid of it.  The less code the better.

kalefranz commented on conda/cli/activate.py:
```diff
@@ -150,13 +151,18 @@ def main():
             if e.errno == errno.EPERM or e.errno == errno.EACCES:
-                sys.exit("Cannot activate environment {}, do not have write access to write conda symlink".format(sys.argv[2]))
+                msg = "Cannot activate environment {}, ".format(sys.argv[2])
+                msg += "not have write access to conda symlink"
```
This string concat is bad for performance.  End up making three copies when you only need one.  Do this instead

    msg = ("Cannot activate environment {0}, "
                "not have write access to conda symlink".format(sys.argv[2]))

Can also do this

    msg = ("Cannot activate environment {0}, not have write access to conda symlink"
                .format(sys.argv[2]))

Also note making the index `{0}` explicit.  Keeps compatibility with python 2.6 (even though we don't explicitly support it) and it also performs *much* faster.

kalefranz commented on conda/cli/common.py:
```diff
@@ -104,7 +106,7 @@ def add_parser_prefix(p):
         action="store",
         help="Name of environment (in %s)." %
-                            os.pathsep.join(config.envs_dirs),
+             os.pathsep.join(config.envs_dirs),
```
eeks bump this to the previous line

kalefranz commented on conda/plan.py:
```diff
@@ -31,10 +30,10 @@
-                                RM_FETCHED, PREFIX, PRINT, PROGRESS,
-                                SYMLINK_CONDA)
+(FETCH, EXTRACT, UNLINK, LINK, RM_EXTRACTED, RM_FETCHED, PREFIX, PRINT,
+    PROGRESS, SYMLINK_CONDA)
```
I *really* don't like stuff like this.  Doesn't `# NOQA` on the import silence pyflakes?  Let's just do that...

kalefranz commented on conda/plan.py:
```diff
@@ -64,10 +63,10 @@ def display_actions(actions, index, show_channel_urls=None):
 
         if index and len(actions[inst.FETCH]) > 1:
+            num_bytes = sum(index[dist + '.tar.bz2']['size']
+                            for dist in actions[inst.FETCH])
```
Could have been the same line, as far as I'm concerned.  (Also don't mind it being two lines.)  I'm sure you noticed max line length set to 99 in `setup.cfg`.  That was definitely intentional.  Still PEP-8 compliant, but for me, more comfortable than 79.

kalefranz commented on conda/plan.py:
```diff
@@ -483,7 +486,9 @@ def remove_actions(prefix, specs, index, force=False, pinned=True):
             else:
-                sys.exit("Error: this 'remove' command cannot be executed because it\nwould require removing 'conda' dependencies.")
+                msg = "Error: this 'remove' command cannot be executed because it"
+                msg += "\nwould require removing 'conda' dependencies."
```
Same comment as above with the string concat.  Also prefer `\n` being *actually* at the end of a line if it's going to be used.

    msg = ("Error: this 'remove' command cannot be executed because it\n"
           "would require removing 'conda' dependencies.")

kalefranz commented on conda/utils.py:
```diff
@@ -98,22 +98,29 @@ def unix_path_to_win(path, root_prefix=""):
+
+    def _translation(found_path):
+        group = found_path.group(0)
+        return group[len(root_prefix) + 1] + ":" + group[len(root_prefix) + 2:].replace("/", "\\")
```
Here, just because this PR is all about style, I think this is clearer:

    return "{0}:{1}".format(group[len(root_prefix) + 1],
                            group[len(root_prefix) + 2:].replace("/", "\\"))

And now that I'm looking at this, is the style around the `+` signs actually correct?  I'm guessing it is, but here I would have myself preferred

    return "{0}:{1}".format(group[len(root_prefix)+1],
                            group[len(root_prefix)+2:].replace("/", "\\"))

`¯\_(ツ)_/¯`

kalefranz commented on conda/utils.py:
```diff
@@ -98,22 +98,29 @@ def unix_path_to_win(path, root_prefix=""):
-cygwin_path_to_win = lambda path : unix_path_to_win(path, "/cygdrive")
+def win_path_to_cygwin(path):
+    return win_path_to_unix(path, "/cygdrive")
+
```
Shouldn't there be two new lines here?  Or is that really only between classes?

kalefranz commented on setup.cfg:
```diff
@@ -13,17 +13,17 @@ markers =
 max-line-length = 99
-ignore = E121,E123,E126,E133,E226,E241,E242,E704
-exclude = build/*,.tox/*,tests/*,ve/*,*/_vendor/*,conda/auxlib/*,conda/progressbar/*
+ignore = E121,E123,E126,E133,E226,E241,E242,E302,E704
```
Ahh. `E302`.  Let's leave this here for now. And then remove it at some point in the future.

kalefranz commented on tests/test_activate.py:
```diff
@@ -225,16 +225,6 @@ def _format_vars(shell):
-    assert_equals(test_cygwin_path, win_path_to_cygwin(test_win_path))
-    assert_equals(test_win_path, cygwin_path_to_win(test_cygwin_path))
-
-
```
Why remove this?  Not saying it's wrong to remove it.  Just don't understand...

kalefranz commented:
> Not sure what's going on with the Appveyor build though...

It's an error in master.  Didn't realize it was there.

Also, you're definitely my favorite person of the day.

alanhdu commented:
Addressed code review and merged the master branch (I can rebase if you think that's nicer).

> Nope. If it's unused, (this repo's) tests pass, and nobody raises an objection with the code review, get rid of it. The less code the better.

I'll make a separate PR to remove some dead code then. Other than `conda-env`, `conda-build` and `constructor`, are there any other projects I should check to make sure I'm not deleting used code?

Glad you think this is helpful! Conda's a god-send for me (especially on Windows -- no more weird mingw-msvc incompatibility errors), so I'm definitely happy to help improve it.

kalefranz commented:
:tada: 

If you ever wanted to do this for the conda-build repo too... 😆
